entity-service-client/src/main/java/org/hypertrace/entity/data/service/client/EdsCacheClient.java

```diff
@@ -224,4 +226,71 @@ public void upsertEnrichedEntities(String tenantId, EnrichedEntities enrichedEnt
   public void upsertRelationships(String tenantId, EntityRelationships relationships) {
     client.upsertRelationships(tenantId, relationships);
   }
+
+  public void updateBasedOnChangeEvent(
+      EntityChangeEventKey entityChangeEventKey, EntityChangeEventValue entityChangeEventValue) {
+    LOG.debug("Entity change event is {}, {} ", entityChangeEventKey, entityChangeEventValue);
+
+    switch (entityChangeEventValue.getEventCase()) {
+      case CREATE_EVENT:
+        // ignore create events, don't populate caches if not necessary
+        break;
+      case UPDATE_EVENT:
+        updateCacheValues(
+            entityChangeEventKey, entityChangeEventValue.getUpdateEvent().getLatestEntity());
+        break;
+      case DELETE_EVENT:
+        getEntityCacheKeys(entityChangeEventKey)
+            .forEach(cacheKey -> entityCache.invalidate(cacheKey));
+        entityIdsCache.invalidate(
+            getIdsCacheKey(
+                entityChangeEventKey, entityChangeEventValue.getDeleteEvent().getDeletedEntity()));
+        break;
+      default:
+        LOG.warn(
+            "Entity change event value has invalid event type -> {}",
+            entityChangeEventValue.getEventCase());
+    }
+  }
+
+  private void updateCacheValues(EntityChangeEventKey entityChangeEventKey, Entity entity) {
```

Comment:
I was thinking the whole method would be common between update and delete. If that seems painful to do, I would at least make update and delete consistent since they're both basically doing the same thing but one is inline in switch and the other is broken out into its own method (which I would argue is preferable)